code/modules/projectiles/guns/energy/laser.dm

```diff
@@ -10,14 +10,10 @@
 	shaded_charge = 1
 	modifystate = TRUE
 
-/obj/item/gun/energy/laser/Initialize()
-	..()
-	AddComponent(/datum/component/extralasers)
-
 /obj/item/gun/energy/laser/attackby(obj/item/I, mob/user, params)
-	..()
+	. = ..()
 	if(modifystate && istype(I, /obj/item/crowbar))
-		SEND_SIGNAL(src,COMSIG_DETACH_LENS)
+		SEND_SIGNAL(src, COMSIG_PARENT_ATTACKBY)

```

Comment:
We already have an attackby signal, you need to use that. Do not make a new signal here of any kind